conjure-java-core/src/main/java/com/palantir/conjure/java/types/UnionGenerator.java

```diff
@@ -173,6 +174,40 @@ private static MethodSpec generateGetValue(ClassName baseClass) {
                     return builder.build();
                 })
                 .collect(Collectors.toList());
+        staticFactories.add(generateUnknownStaticFactory(unionClass, memberTypeDefs));
+        return staticFactories;
+    }
+
+    private static MethodSpec generateUnknownStaticFactory(ClassName unionClass, List<FieldDefinition> memberTypeDefs) {
+        String typeParam = "type";
+        String valueParam = "value";
+        MethodSpec.Builder builder = MethodSpec.methodBuilder("unknown")
+                .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
+                .addParameter(String.class, typeParam)
+                .addParameter(Object.class, valueParam)
+                .returns(unionClass);
+        // begin switch statement
+        builder.beginControlFlow("switch($L)", Expressions.requireNonNull("type", "Type is required"));
+        // add all cases
+        memberTypeDefs.forEach(memberTypeDef -> {
+            String memberName = sanitizeUnknown(memberTypeDef.getFieldName()).get();
```

Comment:
This doesn't need to be sanitized since it's the raw wire string

Reply:
ah makes sense, so we'd want users to call `.unknown("unknown", <thing>)` since `unknown` is the type in the spec